packages/ember-core/src/components/form/search.gts

```diff
@@ -0,0 +1,424 @@
+import { on } from '@ember/modifier';
+import { action, get } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { restartableTask, timeout } from 'ember-concurrency';
+// @ts-expect-error Ember keyboard doesn't currently ship a type for the `on-key` modifier
+// https://github.com/adopted-ember-addons/ember-keyboard/issues/464
+import onKey from 'ember-keyboard/modifiers/on-key';
+
+import BoundValue from './bound-value.ts';
+import onClickOutside from '../../modifiers/on-click-outside.ts';
+import onInsert from '../../modifiers/on-insert.ts';
+
+import type { Optional } from '../../';
+
+declare type SearchOption<T> = {
+  label: string;
+  value: string | T;
+  raw: T;
+};
+
+interface SearchItemSignature<T> {
+  Args: {
+    option: SearchOption<T>;
+    index: number;
+    activeIndex?: number;
+  };
+  Element: HTMLLIElement;
+}
+
+class SearchItem<T> extends Component<SearchItemSignature<T>> {
+  @tracked
+  active = true;
+
+  get classList() {
+    const classes = ['dropdown-item'];
+
+    if (this.isActive) {
+      classes.push('active');
+    }
+
+    return classes.join(' ');
+  }
+
+  get isActive() {
+    return this.args.activeIndex === this.args.index;
+  }
+
+  <template>
+    <li
+      class={{this.classList}}
+      role="option"
+      aria-selected={{this.isActive}}
+      ...attributes
+    >{{@option.label}}</li>
+  </template>
+}
+
+export interface SearchSignature<T> {
+  Args: {
+    basic?: boolean;
+    clearable?: boolean;
+    describedBy?: string;
+    disabled?: boolean;
+    displayPath?: string;
+    format?: ((value: Optional<string>) => string) | false;
+    hideSearchIcon?: boolean;
+    id?: string;
+    isInvalid?: boolean;
+    isWarning?: boolean;
+    loading?: boolean;
+    minCharacters?: number;
+    noResultsLabel?: string;
+    placeholder?: string;
+    query: (searchString: string) => Promise<T[]>;
+    readonly?: boolean;
+    scrollable?: boolean;
+    searchTimeout?: number;
+    serializationPath?: string;
+  };
+  Element: HTMLDivElement;
+}
+
+export default class Search<T> extends BoundValue<
+  SearchSignature<T>,
+  string | T
+> {
+  @tracked
+  activeIndex = -1;
+
+  @tracked
+  options: T[] = [];
+
+  @tracked
+  isFocused = false;
+
+  @tracked
+  menuElement: Optional<HTMLElement> = null;
+
+  @tracked
+  searchInputElement: Optional<HTMLElement> = null;
+
+  @tracked
+  searchString = '';
+
+  get clearable() {
+    if (this.args.basic) {
+      return false;
+    }
+
+    return this.args.clearable ?? false;
+  }
+
+  get hideSearchIcon() {
+    if (this.args.basic) {
+      return true;
+    }
+
+    return this.args.hideSearchIcon ?? false;
+  }
+
+  get loading() {
+    return this.args.loading || this.onQuery.isRunning;
+  }
+
+  get minCharacters() {
+    return this.args.minCharacters ?? 1;
+  }
+
+  get noResultsLabel() {
+    return this.args.noResultsLabel ?? 'No results found';
```

Comment:
This needs to be considered [a translatable string](https://[ember-intl](https://ember-intl.github.io/ember-intl/docs/services/intl-part-1#t-).github.io/ember-intl/docs/services/intl-part-1#t-) and treated appropriately - I would recommend something like this:
```ts
return this.args.noResultsLabel ?? this.intl.t('<some-key>');
```